fdbserver/RestoreMaster.actor.cpp

```diff
@@ -0,0 +1,1339 @@
+						curFileOffset = 0;
+						continue;
+					}
+					if ( loadSizeB / sampleB >= sampleIndex ) {
+						break;
+					}
+					if (curFileIndex >= self->files.size()) {
+						break;
+					}
+					loadSizeB += std::min( self->files[curFileIndex].blockSize, std::max(self->files[curFileIndex].fileSize - curFileOffset * self->files[curFileIndex].blockSize, (int64_t) 0) );
+					curFileOffset++;
+					if ( self->files[curFileIndex].blockSize == 0 || curFileOffset >= self->files[curFileIndex].fileSize / self->files[curFileIndex].blockSize ) {
+						curFileOffset = 0;
+						curFileIndex++;
+					}
+				}
+				if ( curFileIndex >= self->files.size() ) {
+					allLoadReqsSent = true;
+					break;
+				}
+
+				//sampleIndex++;
+
+				// Notify loader to sample the file
+				LoadingParam param;
+				param.url = request.url;
+				param.version = self->files[curFileIndex].version;
+				param.filename = self->files[curFileIndex].fileName;
+				param.offset = curFileOffset * self->files[curFileIndex].blockSize; // The file offset in bytes
+				//param.length = std::min(self->files[curFileIndex].fileSize - self->files[curFileIndex].cursor, loadSizeB);
+				param.length = std::min(self->files[curFileIndex].blockSize, std::max((int64_t)0, self->files[curFileIndex].fileSize - param.offset));
+				loadSizeB += param.length;
+				sampleIndex = std::ceil(loadSizeB / sampleB);
+				curFileOffset++;
+
+				//loadSizeB = param.length;
+				param.blockSize = self->files[curFileIndex].blockSize;
+				param.restoreRange = restoreRange;
+				param.addPrefix = addPrefix;
+				param.removePrefix = removePrefix;
+				param.mutationLogPrefix = mutationLogPrefix;
+				if ( !(param.length > 0  &&  param.offset >= 0 && param.offset < self->files[curFileIndex].fileSize) ) {
+					printf("[ERROR] param: length:%ld offset:%ld fileSize:%ld for %ldth file:%s\n",
+							param.length, param.offset, self->files[curFileIndex].fileSize, curFileIndex,
+							self->files[curFileIndex].toString().c_str());
+				}
+
+
+				printf("[Sampling][File:%ld] filename:%s offset:%ld blockSize:%ld filesize:%ld loadSize:%ldB sampleIndex:%ld\n",
+						curFileIndex, self->files[curFileIndex].fileName.c_str(), curFileOffset,
+						self->files[curFileIndex].blockSize, self->files[curFileIndex].fileSize,
+						loadSizeB, sampleIndex);
+
+
+				ASSERT( param.length > 0 );
+				ASSERT( param.offset >= 0 );
+				ASSERT( param.offset <= self->files[curFileIndex].fileSize );
+
+				printf("[Sampling][CMD] Node:%s Loading %s on node %s\n", 
+						self->describeNode().c_str(), param.toString().c_str(), loaderID.toString().c_str());
+
+				self->cmdID.nextCmd(); // The cmd index is the i^th file (range or log file) to be processed
+				if (!self->files[curFileIndex].isRange) {
+					cmdType = RestoreCommandEnum::Sample_Log_File;
+					self->cmdID.setPhase(RestoreCommandEnum::Sample_Log_File);
+					cmdReplies.push_back( loaderInterf.sampleLogFile.getReply(RestoreLoadFileRequest(self->cmdID, param)) );
```

Comment:
Given that they have the same parameters, return value, and do the same conceptual thing, I'd probably just make this one sampleFile, and do the log vs range branch on the loader side?  Does keeping them separate here make the phase distinction clearer?  (Is there a phase distinction?)